### DISCUSSION TITLE
flet-extend

### DISCUSSION THREAD

**Discussion by ssseee66** [flet-dev/flet#3792] [Q&A]
I extended the flet application plug-in with a flutter plug-in I wrote to listen for PDA broadcast events, and the TextField control returned in my flutter package < package _ name >.dart. Now I add the returned control to the flet application via a Python control, It works fine, but I don't know how to get the control and the control properties, like TectField.text (returning the control as TextField in the Flutter package) and do the next data processing. I hope you can help me, thank you!!
I am a Chinese, English is not very good, I hope you can understand my meaning and appeal.

**Comment by ssseee66** [comment]
# pda_listener.dart

import 'package:flet/flet.dart';
import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';
import 'package:my_pda_scanner/my_pda_scanner_mixin.dart';

class PdaListener extends StatefulWidget {
  Control? parent;
  late Control control;

  PdaListener({
    super.key,
    required this.parent,
    required this.control,
  });

  @override
  _PdaListener createState() => _PdaListener();
}

class _PdaListener extends State<PdaListener>
    with MyPdaScannerMixin<PdaListener> {
  String data_code = "NULL";

  @override
  Widget build(BuildContext context) {
    var pda_code = widget.control.attrString("pda_code", data_code);
    TextEditingController _controller = TextEditingController.fromValue(
        TextEditingValue(text: "扫描到数据:$data_code"));
    Widget pda_control = TextField(controller: _controller);
    return constrainedControl(
        context, pda_control, widget.parent, widget.control);
  }

  @override
  Future<void> myPdaScannerCodeHandle(String code) async {
    /// 编写你的逻辑
    setState(() {
      data_code = code;
    });
  }
}


# pda_listener.py

from flet_core.constrained_control import ConstrainedControl
from typing import Any, Optional
import flet as ft


class PdaListener(ConstrainedControl):
    """
    PdaListener 控件。
    """

    def __init__(self, pda_code = None):
        ConstrainedControl.__init__(self)
        self.pda_code = pda_code
    


    def _get_control_name(self):
        return "pda_listener"
    
    def before_update(self):
        super().before_update()
        self.pda_code = self._get_attr("pda_code")
       
    @property
    def pda_code(self):
        print("ksjsksj")
        print(self._get_attr("pda_code"))

        return self._get_attr("pda_code")
    @pda_code.setter
    def pda_code(self, value):
        self._set_attr("pda_code", value)
    

# main.py

import flet as ft
from extend.python.controls.pda_listener import PdaListener
def main(page: ft.Page):
    pda_listener = PdaListener()

    page.add(ft.Text("Hello Flet!!!!!"))
    page.add(pda_listener)
    
ft.app(main)



This is my main file, and I want to get the text value of the PdaListener() control in main.py

**Comment by ndonkoHenri** [comment]
Can you share your code as a repo? I could help you then.

**Comment by ssseee66** [comment]
I'm very sorry for the time difference between us, so it took us so long to reply you. I've put my code in the github repository at https://github.com/ssseee66/my_app.git, thank you for your reply.
________________________________
发件人: TheEthicalBoy ***@***.***>
发送时间: 2024年8月13日 4:48
收件人: flet-dev/flet ***@***.***>
抄送: 略略略 ***@***.***>; Author ***@***.***>
主题: Re: [flet-dev/flet] flet-extend (Discussion #3792)


Can you share your code as a repo? I could help you then.

―

**Reply by ndonkoHenri** [reply]
Thanks. 
Now can you please detaily explain the issue? I honestly didnt get you completely.

**Comment by ssseee66** [comment]
I added an instance of a Python control called PdaListener to the page, built as an apk that can listen to broadcasts on the PDA mobile terminal, and now I want to get this text value for the next step.
How do I get the text value in the input field on the image, which is the control that I have been extended to get
________________________________
发件人: TheEthicalBoy ***@***.***>
发送时间: 2024年8月13日 20:19
收件人: flet-dev/flet ***@***.***>
抄送: 略略略 ***@***.***>; Author ***@***.***>
主题: Re: [flet-dev/flet] flet-extend (Discussion #3792)


Thanks.
Now can you please detaily explain the issue? I honestly didnt get you completely.

―
